docs/tutorial.md

```diff
@@ -32,13 +32,118 @@ require('vendor/autoload.php');
 TODO: create client instance with discovery and do some requests
 ```
 
-## Handling errors
+## Using an asynchronous client
 
-TODO: explain how to handle exceptions, distinction between network exception and HttpException.
+When using an asynchronous client, it will use a PSR-7 `RequestInterface` and returns a `Http\Client\Promise` :
+
+```php
+$httpAsyncClient = new HttpAsyncClientImplementation();
```

Comment:
then the tutorial should use it i think, instead of this line which can't be executed in reality (you would need to know your real client/adapter class name and use that instead)